### ISSUE TITLE
Update Operator to link service accounts if a given secret 'syndesis-pull-secret' exists

### ISSUE THREAD

**Issue by rhuss** [syndesisio/syndesis#4900]
## Minimal required

The operator should, after he installed the templates at https://github.com/syndesisio/syndesis/blob/fc7ad4c66d819759e24b3f293de9887e981eb81f/install/operator/pkg/syndesis/action/install.go#L89 link the service accounts if a given secret in the operator namespace exist.

**The secret will be called "syndesis-pull-secret".** 

## Optional enhancements 

One could go one step further to add a boolean field to the CRD like 'enablePullAuthentication' which could be used to make this behaviour more explicit. I.e. if this flag is set but the secret is missing, that an error is thrown. If this flag is not set and the secret exists, the secret is ignored.

I suggest implementing this behaviour for 7.4+.

Also one could make this kind of SA update more dynamic and e.g. monitor the CR all the time and when this flag changes in the CR then SA gets adapted. This should be implemented when the CR update functionality is implemented via #3422 .

An advantage of this post-adaption approach (create templates with SAs, modifying SAs afterwards) would be, that the template would not have to be adapted and can be directly used again on OpenShift Online without further change.

// cc @heiko-braun @avano

**Comment by rhuss** [comment]
@avano According to https://access.redhat.com/RegistryAuthentication (see last example on the page) I think that we have to attach the secret to the "builder" SA. Can you try a 

```
oc secrets link builder <pull_secret_name>
```

and then try again the deployment of the runtime image ?

**Comment by avano** [comment]
@rhuss I tried that and it didn't work for me. I also tried to create some sample bc manually (that uses syndesis-s2i image from registry.redhat.io) to see if there may be some problem in how syndesis creates the bc, but that wasn't successful either

**Comment by avano** [comment]
just for the record, I also tried to add it everywhere if something helps, but it didnt':

```
for sa in "builder" "default" "deployer" "syndesis-integration" "syndesis-oauth-client" "syndesis-prometheus" "syndesis-server"; do oc secrets link $sa <pull-secret-name> --for=pull; done
```

**Comment by astefanutti** [comment]
For the `builder` SA, the `--for=pull` option should be removed.

**Comment by avano** [comment]
@astefanutti you're right, it works without the `--for-pull`

**Comment by astefanutti** [comment]
@avano great, thx for the quick feedback!

**Comment by heiko-braun** [comment]
Great, than we can bake it right into the changes we do atm. Thanks @astefanutti and @avano !

**Comment by astefanutti** [comment]
@heiko-braun yes let's put all that into the operator code.

**Comment by astefanutti** [comment]
It should be fixed with #4951. I've tested the fix by checking the service accounts do have the image pull secret linked. It should ideally be tested against a registry requiring authentication.

**Comment by avano** [comment]
works ok with 1.6.3
